Rebus.AdoNet/Dialects/TypeNames.cs

```diff
@@ -1,176 +1,182 @@
-
-			return Get(typecode);
-		}
-
-		/// <summary>
-		/// Set a type name for specified type key and capacity
-		/// </summary>
-		/// <param name="typecode">the type key</param>
-		/// <param name="capacity">the (maximum) type size/length</param>
-		/// <param name="value">The associated name</param>
-		public void Put(DbType typecode, uint capacity, string value)
-		{
-			SortedList<uint, string> map;
-			if (!weighted.TryGetValue(typecode, out map))
-			{
-				// add new ordered map
-				weighted[typecode] = map = new SortedList<uint, string>();
-			}
-			map[capacity] = value;
-		}
-
-		/// <summary>
-		/// 
-		/// </summary>
-		/// <param name="typecode"></param>
-		/// <param name="value"></param>
-		public void Put(DbType typecode, string value)
-		{
-			defaults[typecode] = value;
-		}
-	}
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Data;
+using System.Text;
+
+namespace Rebus.AdoNet.Dialects
+{
+	/// <summary>
+	/// This class maps a DbType to names.
+	/// </summary>
+	/// <remarks>
+	/// Associations may be marked with a capacity. Calling the <c>Get()</c>
+	/// method with a type and actual size n will return the associated
+	/// name with smallest capacity >= n, if available and an unmarked
+	/// default type otherwise.
+	/// Eg, setting
+	/// <code>
+	///     Names.Put(DbType,           "TEXT" );
+	///     Names.Put(DbType,   255,    "VARCHAR($l)" );
+	///     Names.Put(DbType,   65534,  "LONGVARCHAR($l)" );
+	/// </code>
+	/// will give you back the following:
+	/// <code>
+	///     Names.Get(DbType)           // --> "TEXT" (default)
+	///     Names.Get(DbType,100)       // --> "VARCHAR(100)" (100 is in [0:255])
+	///     Names.Get(DbType,1000)  // --> "LONGVARCHAR(1000)" (100 is in [256:65534])
+	///     Names.Get(DbType,100000)    // --> "TEXT" (default)
+	/// </code>
+	/// On the other hand, simply putting
+	/// <code>
+	///     Names.Put(DbType, "VARCHAR($l)" );
+	/// </code>
+	/// would result in
+	/// <code>
+	///     Names.Get(DbType)           // --> "VARCHAR($l)" (will cause trouble)
+	///     Names.Get(DbType,100)       // --> "VARCHAR(100)"
+	///     Names.Get(DbType,1000)  // --> "VARCHAR(1000)"
+	///     Names.Get(DbType,10000) // --> "VARCHAR(10000)"
+	/// </code>
+	/// </remarks>
+	public class TypeNames
+	{
+		public const string LengthPlaceHolder = "$l";
+		public const string PrecisionPlaceHolder = "$p";
+		public const string ScalePlaceHolder = "$s";
+
+		private readonly Dictionary<DbType, SortedList<uint, string>> weighted = new Dictionary<DbType, SortedList<uint, string>>();
+		private readonly Dictionary<DbType, string> defaults = new Dictionary<DbType, string>();
+
+		/// <summary>
+		/// Expose defaults as a readonly collection for testing purpose..
+		/// </summary>
+		internal IReadOnlyDictionary<DbType, string> Defaults => new ReadOnlyDictionary<DbType, string>(defaults);
```

Comment:
What are 'defaults' in this context? I dont really see a need to add this here.

Reply:
will remove this; was for testing purpose.